Add documentation of MID wires (#167)
FPGA-Research-Manchester/FABulous: docs/source/fabric_definition.rst

  In summary, the example shows how a termination tile can be used to provide more complex interface blocks and all this can be easily modelled and implemented with FABulous.
  
  .. note::  The ``destination_name`` is refering to the port name used at the destination tile. FABulous will throw an error if the destination tile does not provide that port name.

>   Aside from ``BEGIN`` and ``END``, there also exist ``MID`` ports, which are used for wires spanning over two tiles.

Reviewer: This phrasing might be slightly less ambiguous?

```suggestion
  Aside from ``BEGIN`` and ``END``, there also exist ``MID`` ports, which can be used for wires spanning more than two tiles.
```

Author: Yeah seems better! Thanks!

  Although they route over two tiles, they also have a tap on the middle tile.
  On the middle tile, the sink is called ``MID`` while the source is still a ``BEG`` port with the addition of ``b``.
  In the example below this is illustrated. The wire on the left goes from ``E2BEG0`` to ``E2MID`` which is the tap on the middle tile.

>   Inside the switch matrix, ``E2MID`` and ``E2BEGb`` are connected. The left wire the starts at ``E2BEGb`` and ends at ``E2END``.

Reviewer: ```suggestion
  Inside the switch matrix, ``E2MID`` and ``E2BEGb`` are connected. The left wire then starts at ``E2BEGb`` and ends at ``E2END``.
```

Author: Thanks for catching this!
